File: Plantão Facil/src/Domain/Entities/Escala.cs
namespace Domain.Entities;

public class Escala : Base
{
    public DateTime DataInicial  { get; set; }
    
    public DateTime DataTermino { get; set; }
    public long FuncionarioId { get; set; }
    public List<Funcionario> Funcionarios { get; set; } = null!;
    public long VagaId { get; set; }
    public List<Vaga> Vagas { get; set; } = null!;
    
    public override bool Validate()
    {
        return true;
    }
}

File: Plantão Facil/src/Mobile/src/app/models/Escala.ts
import { Funcionario } from "./Funcionario";
import { Vaga } from "./Vaga";

export class Escala {
    public id: number;
    public dataInicial: Date;
    public dataTermino: Date;
    public funcionarioId: number;
    public vagaId: number;
    public funcionarios: Funcionario[]; 
    public vagas: Vaga[]; 

    constructor() {
        this.id = 0;
        this.dataInicial = new Date();
        this.dataTermino = new Date();
        this.funcionarioId = 0;
        this.vagaId = 0;
        this.funcionarios = [];
        this.vagas = [];
    }
}
